### PR TITLE
E_CLASSROOM-212 [FE] Update Password Reset Validation

### PR DESCRIPTION
**Pull request by Teraaay**
### Issue Link

https://framgiaph.backlog.com/view/E_CLASSROOM-212

### Definition of Done
- [x] User will be displayed a validation message same as signup page if there is something wrong with input

### Commands to Run
N/A

#### Main note
- N/A
#### Pages Affected
- http://localhost:3003/reset-password

### Scenarios/ Test Cases
- [x] it should `display a validation error message` when `the user entered a wrong email` 
#### User Interface
- [x] when `the user enters a wrong email the input field` should ` have a red highlight and should display the specific error under it`
### Screenshots
image

### CODE REVIEW

**Comment by catampongan** [comment]
### Working
- [x] Display validation message if the email is incorrect/invalid
- [x] Able to apply red outline color

### Comment
-  The buttons should be removed if the sending of recovery link is successful

----- 

image

reset-password

-----
- Total Test Time: 0.5 hr

**Comment by catampongan** [comment]
### Approved PR #72 

- [x] Display validation message if the email is incorrect/invalid
- [x] Able to apply red outline color
- [x] Able to remove buttons if the sending of recovery link is successful
- [x] Able to redirect login page to avoid immediate request of recovery link 

----- 

image

reset-password

-----
- Total Test Time: 0.75 hr
- Total Iterations: 1
- Total Comments: 1

**Comment by catampongan** [comment]
### Approved PR #72 

- [x] Display validation message if the email is incorrect/invalid
- [x] Able to apply red outline color
- [x] Able to remove buttons if the sending of recovery link is successful
- [x] Able to redirect login page to avoid immediate request of recovery link 

----- 

image

reset-password

-----
- Total Test Time: 0.75 hr
- Total Iterations: 1
- Total Comments: 1

File: src/pages/student/main/PasswordReset/index.js [hunk 1]
```diff
@@ -4,137 +4,118 @@ import Container from 'react-bootstrap/Container';
...
+                    render={({ field: { onChange, value, ref } }) => (
+                      <Form.Control
+                        onChange={onChange}
+                        value={value}
+                        ref={ref}
+                        className='cntrs'
+                        type='email'
+                        isInvalid={error}
+                        required
+                        maxLength={50}
```
Comment (inahsanjuan) [inline src/pages/student/main/PasswordReset/index.js]: For element attributes, and component props, always use double quotes `""`, not `''`
Check throughout your PR for similar issues

```suggestion
                  <Controller
                    control={control}
                    name="email"
                    defaultValue=''
                    render={({ field: { onChange, value, ref } }) => (
                      <Form.Control
                        onChange={onChange}
                        value={value}
                        ref={ref}
                        className="cntrs"
                        type="email"
                        isInvalid={error}
                        required
                        maxLength={50}
```

File: src/pages/student/main/PasswordReset/index.js [hunk 2]
```diff
@@ -4,137 +4,118 @@ import Container from 'react-bootstrap/Container';
...
+                <Form.Group
+                  id={style.Containercentermargin}
+                  className='mb - 3'
+                  controlId='Email'
                 >
-                  <span className={style.textbutton}>Send Recovery Link</span>
-                </Button>
-              </center>
+                  <Form.Label>
+                    <h6 style={{ marginBottom: '0px' }}>Email</h6>
```
Comment (inahsanjuan) [inline src/pages/student/main/PasswordReset/index.js]: Cant we use any utility class for this, something like `mb-0`

File: src/pages/student/main/PasswordReset/index.js [hunk 3]
```diff
@@ -4,137 +4,118 @@ import Container from 'react-bootstrap/Container';
...
+              <div className={style.center} align='start'>
+                <center>
+                  <Form.Label>
+                    {' '}
+                    <h4> Password Reset </h4>
+                  </Form.Label>
+                </center>
+                <Form.Group
+                  id={style.Containercentermargin}
+                  className='mb - 3'
```
Comment (inahsanjuan) [inline src/pages/student/main/PasswordReset/index.js]: is this not a mistake?
```suggestion
                  className='mb-3'
```

File: src/pages/student/main/PasswordReset/index.js [hunk 4]
```diff
@@ -4,137 +4,118 @@ import Container from 'react-bootstrap/Container';
...
-                  onClick={() => {
-                    setShow(false);
-                  }}
-                  id={style.button}
+            {status === false ? (
+              <div className={style.center} align='start'>
+                <center>
+                  <Form.Label>
+                    {' '}
+                    <h4> Password Reset </h4>
```
Comment (inahsanjuan) [inline src/pages/student/main/PasswordReset/index.js]:
```suggestion
                    <h4>Password Reset</h4>
```

File: src/pages/student/main/PasswordReset/index.js [hunk 5]
```diff
@@ -4,137 +4,118 @@ import Container from 'react-bootstrap/Container';
...
+                        Send Recovery Link
+                      </span>
+                    </Button>
+                  ) : (
+                    <Button type='submit' id={style.button} disabled>
+                      <span className={style.textbutton}>
+                        Send Recovery Link
+                      </span>
+                    </Button>
+                  )}
```
Comment (inahsanjuan) [inline src/pages/student/main/PasswordReset/index.js]: No need to have ternary here, difference is only the `disabled` prop
```suggestion
                  <Button type='submit' id={style.button} disabled={submitStatus}>
                    <span className={style.textbutton}>
                      Send Recovery Link
                    </span>
                  </Button>
```

File: src/pages/student/main/PasswordReset/index.module.css [hunk 6]
```diff
@@ -76,18 +76,32 @@
 }
 
 #Btncolor {
-  background-color: #C0D5D8;
-  border-color: #C0D5D8;
+  background-color: #c0d5d8;
+  border-color: #c0d5d8;
```
Comment (inahsanjuan) [inline src/pages/student/main/PasswordReset/index.module.css:80]: Just a confirmation, do we not have global variables for colors?

**Review by inahsanjuan — CHANGES_REQUESTED** [review]
- [x] Please update all prop definition to double quotes `""`

File: src/pages/student/main/PasswordReset/index.module.css (hunk 6 above)
Comment (Teraaay) [reply src/pages/student/main/PasswordReset/index.module.css:80]: We don't have global variables for colors, Ms @inahsanjuan.

File: src/pages/student/main/PasswordReset/index.js [hunk 7]
```diff
@@ -45,96 +41,76 @@ const PasswordReset = () => {
...
-              <center>
-                <div>
-                  <a href="login" className={style.cancel}>
-                    Cancel
-                  </a>
+                <center>
+                  <Button
+                    type="submit"
+                    id={style.button}
+                    className={submitStatus === true ? 'disabled' : ''}
```
Comment (inahsanjuan) [inline src/pages/student/main/PasswordReset/index.js]: no need for `=== true`
```suggestion
                    className={submitStatus ? 'disabled' : ''}
```

Comment (inahsanjuan) [reply src/pages/student/main/PasswordReset/index.js]: Did you purposely remove disabled={submitStatus} prop? i think its also important

**Review by inahsanjuan — CHANGES_REQUESTED** [review]

**Review by inahsanjuan — APPROVED** [review]

**Pull request merged**